wgpu/src/assertions.rs

```diff
@@ -0,0 +1,50 @@
+// TODO: Do not merge without docs
```

Comment:
We should move these into a file in wgpu-types, so the whole codebase can share it.